python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -286,6 +286,16 @@ def skip_on_not_all_parents_updated(
         """
         return SkipOnNotAllParentsUpdatedRule(require_update_for_all_parent_partitions)
 
+    @public
+    @staticmethod
+    def skip_on_required_but_nonexistent_parents() -> "SkipOnRequiredButNonexistentParentsRule":
+        """Skip an asset partition if it depends on parent partitions that do not exist.
```

Comment:
Final nit on cleaning up this docstring, maybe something along the lines of "For example, imagine a downstream asset is time-partitioned, starting in 2022, but has a time-partitioned parent which starts in 2023. This rule will skip attempting to materialize downstream partitions from before the start date of the parent"

Reply:
updated!